miden-lib/asm/miden/tx.masm

```diff
@@ -65,27 +65,26 @@ end
 #! note_idx is the index of the crated note.
 export.create_note
     syscall.create_note
-    # => [note_idx, EMPTY_WORD, 0]
-
-    # clear the padding from the kernel response
-    movdn.4 dropw swap drop
-    # => [note_idx]
+    # => [note_idx, PAD(15)]
 end
 
 #! Adds the ASSET to the note specified by the index.
 #!
-#! Inputs: [note_idx, ASSET]
-#! Outputs: [note_idx]
+#! Inputs:  [ASSET, note_idx]
+#! Outputs: [ASSET, note_idx]
 #!
 #! note_idx is the index of the note to which the asset is added.
 #! ASSET can be a fungible or non-fungible asset.
 export.add_asset_to_note
+    dupw movup.8
+    # => [note_idx, ASSET, ASSET]
+
     syscall.add_asset_to_note
-    # => [note_idx, EMPTY_WORD]
+    # => [note_idx, EMPTY_WORD, ASSET]
 
     # clear the padding from the kernel response
-    movdn.4 dropw
-    # => [note_idx]
+    movdn.8 dropw
+    # => [ASSET, note_idx]
 end
```

Comment:
Could we also update `add_asset_to_note` in `api.masm` to have the same signature as here? This way, we won't need to do extra `dupw` and `dropw` here.